reps/2025-11-21-ray-history-server/2025-11-21-ray-history-server.md

```diff
@@ -0,0 +1,143 @@
+## Ray History Server
+
+### General Motivation
+
+It is becoming increasingly common amongst Ray users to treat Ray clusters as an ephemeral unit of compute
+that only runs for the duration of a single (or multiple) Ray jobs. This pattern results in significant improvements to
+cost efficiency and resource sharing, especially in capacity constraint environments where hardware accelerators are scarce.
+
+However, a fundamental trade-off to this approach, compared to long-lived interactive Ray clusters, is that users
+lose access to the Ray Dashboard which is often treated as the entrypoint for most observability signals
+in a Ray cluster. While it’s possible to export the relevant signals to external sources, users prefer the experience
+of using Ray Dashboard as a single source of truth to debug job failures.
+
+This enhancement proposal introduces the concept of a Ray History Server, which can orchestrate the reconstruction of the
+Ray Dashboard for even terminated Ray clusters. This will be accomplished by leveraging Ray’s Event Export API to persist
+task/actor/job state, and native components in KubeRay for pushing logs and events to blob store (GCS, S3, etc).
+
+### Should this change be within `ray` or outside?
+
+Some components/libraries, such as the event exporter will be in Ray. Everything else will be hosted in the KubeRay project.
+
+## Stewardship
+
+### Owners
+
+- @KunWuLuan (Alibaba)
```

Comment:
@KunWuLuan @MengjinYan  anyone else from Alibaba or Anyscale we should add as owner here?

Reply:
done